tests/browser/modules/selection-state.spec.js

```diff
@@ -44,6 +44,36 @@ describe('Browser: SelectionState', function() {
 
     });
 
+    describe('given a burn-image-again state', function() {
```

Comment:
Actually, notice the `.clear()` describe block in line 154. At that point, the nested `beforeEach` blocks setup a scenario where an image and a drive are selected, which is just what we need. I'd add a `describe('given the preserveImage option')` inside `describe('.clear()')` and move all the new test cases there.